react/features/base/media/middleware.js

```diff
@@ -68,6 +68,32 @@ function _setRoom({ dispatch, getState }, next, action) {
     const state = getState();
     const { room } = action;
     const roomIsValid = isRoomValid(room);
+    const { error } = state['features/base/conference'];
+    const { audioOnly: previousAudioOnly } = state['features/base/conference'];
+    const { audio, video } = state['features/base/media'];
+
+    let audioMuted, facingMode, videoMuted;
+
+    // startAudioOnly
+    //
+    // FIXME Technically, the audio-only feature is owned by base/conference,
+    // not base/media so the following should be in base/conference.
+    // Practically, I presume it was easier to write the source code here
+    // because it looks like startWithAudioMuted and startWithVideoMuted.
+    //
+    // XXX After the introduction of the "Video <-> Voice" toggle on the
+    // WelcomePage, startAudioOnly is utilized even outside of
+    // conferences/meetings.
+    let audioOnly;
+
+    if (room && error) {
```

Comment:
In what case would room be undefined? _setRoom also clears the room?

Reply:
No, this is a join event, so by now we have the new (equal to old) room name here, but we still have a reference to the error that caused the failure of the previous conference

Reply:
It sounds like room cannot be undefined. If that is the case why check for truthiness?

Reply:
setRoom can be invoked with an undefined or invalid room value, then we go back to the welcome page